server/api.go

```diff
@@ -643,8 +648,10 @@ func (p *Plugin) searchIssues(w http.ResponseWriter, r *http.Request, userID str
 	w.Write(resp)
 }
 
-func getPermaLink(siteUrl string, postId string, currentTeam string) string {
-	return fmt.Sprintf("%v/%v/pl/%v", siteUrl, currentTeam, postId)
+func (p *Plugin) getPermaLink(postId string) string {
+	siteUrl := *p.API.GetConfig().ServiceSettings.SiteURL
```

Comment:
While there is a theoretical possibility for `SiteURL` to be `nil` the server does ensure that there are no `nil` values. I don't think it's worth the error handling here.

Reply:
> the server does ensure that there are no nil values

@hanzei is this check for nil values by the server new functionality? We have experienced panics for this same reason in the past and have since guarded against it in most plugins.

Reply:
I was under the impression that this should not be an issue for every server version. Do you have a link to a ticket/conversation where this issue came up?

Reply:
@hanzei Here's the PR that fixed the issue in the Jira plugin https://github.com/mattermost/mattermost-plugin-jira/pull/316

Reply:
Thanks for the hint. I added a check to `OnActivate`.